Guide/authentication.markdown

```diff
@@ -259,6 +260,22 @@ You can also access the user using `currentUser` inside your views:
 |]
 ```
 
+## Performing actions after login
+
+The sessioncontroller has a convenient beforeLogin which is run on login after the user is authenticated, but before the target page is rendered. This can be useful for updating last login time, number of logins etc. Add code for it in your `Web/Controller/Sessions.hs`
+
+```haskell
+instance Sessions.SessionsControllerConfig User where
+    beforeLogin = updateLoginHistory
+
+updateLoginHistory user = do
+    let n = get #logins user
+    user
+        |> set #logins (n+1)
+        |> updateRecord
+    pure ()
+```
+
```

Comment:
Maybe we could also add a second example for the `Disallow login until user is confirmed` case here. We already have that inside the API docs, so basically it's just copy paste :)

https://ihp.digitallyinduced.com/api-docs/IHP-AuthSupport-Controller-Sessions.html#v:beforeLogin

Reply:
Yes! I looked for that but did not find it. I remembered you had an example with blocking login until user has confirmed his email. Will add that later today.